TrueDialog.Client.Standard/Client/TrueDialogClient.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Concurrent;
+
+using System.Reflection;
+
+using TrueDialog.Configuration;
+using TrueDialog.Context;
+
+namespace TrueDialog
+{
+    public class TrueDialogClient : ITrueDialogClient
+    {
+        private readonly IApiCaller m_api;
+
+        internal TrueDialogClient(IApiCaller api)
+        {
+            m_api = api;
+        }
+
+        public TrueDialogClient(ITrueDialogConfigProvider configFactory)
+        {
+            m_api = new ApiCaller(configFactory);
+        }
+
+        public int AccountId { get { return m_api.AccountId; } }
+
+        public void SetAccountId(int accountId)
+        {
+            m_api.AccountId = accountId;
+        }
+
+        #region Contexts
+
+        private readonly ConcurrentDictionary<string, BaseContext> m_contextDictionary = new ConcurrentDictionary<string, BaseContext>();
+        private readonly object m_mutex = new object();
+
+        internal TContext GetContext<TContext>()
```

Comment:
Replace with DI, don't think the Microsoft DI has the nice auto resolve of Lazy<> like Stashbox does though. :/